Fixes for handing of popcnt and for compilation on non-x86 architectures

jdart1 opened the pull request:
Always use TB_CUSTOM_POP_COUNT if defined. If not defined and 
if also TB_NO_HW_POP_COUNT is not defined, use a x86 hardware
instruction, but only if on a recognized compiler/architecture
that has one. Fixes issue #9 (ARM compatibility).

jdart1 committed:
Always use TB_CUSTOM_POP_COUNT if defined. If not defined and
if also TB_NO_HW_POP_COUNT is not defined, use a x86 hardware
instruction, but only if on a recognized compiler/architecture
that has one. Fixes issue #9 (ARM compatibility).

jdart1 commented:
Is there a reason not to merge this? I think without this pull request the code is not really portable to non-x86 architectures. But I am open to other ways of fixing that.

jdart1 committed:
Fix issues found by Coverity Scan:
1. Check return value from fstat.
2. Use strncpy not strcpy in constructing file path.

jdart1 committed:
Set precomp, data, and mapping fields to NULL after free.
Write error messages to stderr, not stdout.
Add error checking for unmap function.

jdart1 committed:
Fix compiler warnings. Fix logic in sanity check in tb_init_impl.

basil00 commented:
Sorry about that.  The change did require me to check some things, which I never got around to. There is a backlog of changes needed for Fathom that I currently have little time for.

jdart1 committed:
Make oppdiag array explicitly signed (fixes compile issue with clang for ARM target).
Update copyright and license info.

jdart1 committed:
Use atomic variable instead of explicit memory barrier if compiling
under C++ 11. Add some asserts.

jdart1 committed:
Use a define to select use of atomic variables. Don't rely on
__cplusplus value to select becauase that is unreliable for MSVC.

jdart1 committed:
Add check in is_legal for king bitmap empty.

jdart1 committed:
Use angle-bracket include for tbconfig.h. This facilitates
overridding the config file by including it from another directory.

jdart1 committed:
Don't make locking code depend on preprocessor variable TB_HAVE_THREADS:
it was not being defined. Use std::mutex if C++11 compile. Update
copyright.

jdart1 committed:
Don't include lock code if TB_NO_THREADS is defined.

jdart1 committed:
Fix asserts in encode_piece (assert was tripping on 6-man tbs: reported by
Andrew Grant).

jdart1 committed:
add std namespace to atomic declarations

jdart1 committed:
Remove size limitation on file paths.

jdart1 committed:
update copyrights

jdart1 committed:
Consistently use TB_USE_ATOMIC define to select usage of atomic variables.

jdart1 committed:
Fix issue 22 in basil00/Fathom/: properly disambiguate moves in
move_to_str. Note: this still does not output complete SAN, because
no indication of check or mate.

jdart1 committed:
Fix compilation when TB_CUSTOM_LSB is not defined. The previous
code assumed gcc on x86_64. Now we use builtins if possible and
otherwise a De Bruijn multiplication implementation.

jdart1 committed:
Remove use of strdup (not defined in C99). Update copyright.

jdart1 committed:
remove reference to binaries

jdart1 committed:
first cut at 7man support (from Cfish)

jdart1 committed:
Fix WDL value when a mate in 1 is possible at the 50-move threshold
(fix ported from Cfish - see https://github.com/syzygy1/Cfish/commit/d4d1493a487ff3804c198ec2032b4bd91bec9088#diff-98bbf44ad0a296906914322f9b11cd63).

jdart1 committed:
Bug fixes and code cleanup. Changes include:
1. Expose new interfaces for DTZ and WDL probing with ranks/scores
as in Cfish, as well as the old Fathom interface.
2. Put both tbprobe.c and tbchess.c under MIT license.
3. Require atomic support, remove this option from tbconfig.h.
4. Use stdendian for big/little endian support.
5. Add constants for mate scores, pawn scores etc. to tbconfig.h.

Note: two test cases from Arasan unit tests still not passing due
to Cfish bug.

jdart1 committed:
remove trailing whitespace

jdart1 committed:
Fix test for KvK. Fix a signed/unsigned compare warning.

jdart1 committed:
optimize by default again

jdart1 committed:
Fixes for C++ and Windows/MSVC compilation.

jdart1 committed:
Fix gcc warnings. Return error if castling status != 0.

jdart1 committed:
Fix Windows error reporting: use correct printf specifier, call GetLastError if CreateFileMapping fails

jdart1 committed:
Merge branch '7man'

jdart1 committed:
update README

jdart1 committed:
small README fix

jdart1 committed:
add some asserts

jdart1 committed:
Fix mingw64 warnings (reported by Vivien Clauzon).

jdart1 committed:
minor README changes

jdart1 committed:
Place PieceType and other definitions within an anonymous C++ namespace
(fixes ODR warning when compiling Arasan with Fathom).

jdart1 committed:
fix some warnings

jdart1 committed:
support MSVC compilation in Unicode mode (-D_UNICODE).

jdart1 committed:
Pass in correct length of Unicode buffer to mbstowcs_s

jdart1 committed:
some edits to history/credits section. Add some further advice
about overriding tbconfig.h.

jdart1 committed:
Fix typo. Add mention that chess code was MIT-licensed.

jdart1 committed:
restore original copyright from Ronald de Man from source tbprobe.c

jdart1 committed:
Bug fix: ensure TB_LARGEST is set zero if Fathom is re-initialized with
an empty string or "<empty>".

jdart1 committed:
update copyrights in LICENSE file

jdart1 committed:
support Emscripten (emcc) compile

jdart1 committed:
mark unreferenced functions as unused, if using C++17 or above

Tearth committed:
Fix false munmap error

jdart1 committed:
Merge pull request #4 from Tearth/munmap

Fix false munmap error

jdart1 committed:
Add option to use __builtin_popcount, as suggested in PR#5.

Matthies committed:
Fix mapping offset for wide dtz tables.

Matthies committed:
Murphys law strikes again, I edited the wrong branch. This fixes it.

jdart1 committed:
Merge pull request #6 from Matthies/master

Fix mapping offset for wide dtz tables.

srogatch committed:
Fix the compilation for C++20

jdart1 committed:
Merge pull request #7 from srogatch/users/srogatch/fix/C++20

Fix the compilation for C++20

jdart1 committed:
Change sprintf to snprintf (fixes clang compiler warnings). Update copyright.

jdart1 committed:
Use madvise to avoid prefetch from tb memory-mapped file region.
Also use FILE_FLAG_RANDOM_ACCESS for opening Windows files, as
in Stockfish.

jdart1 committed:
fix a shadow warning

jdart1 committed:
update copyright dates

jdart1 committed:
Update README to reference LICENSE file.

jdart1 committed:
Fix C++20 warning - don't call init to initialize atomic variables, use the constructor

jdart1 committed:
use more portable initialization for ready array

jdart1 committed:
call atomic_init for C++ versions below 20

spinojara committed:
Add top level Makefile for compiling a shared object

spinojara committed:
Update Makefile

spinojara committed:
Add build instructions

spinojara committed:
Fix typo

jdart1 committed:
Merge pull request #8 from Spinojara/master

Add top level Makefile for compiling a shared object

spinojara committed:
Fix install bug if $(DESTDIR)$(INCDIR) does not exist

jdart1 committed:
Merge pull request #9 from spinojara/master

Fix install bug if $(DESTDIR)$(INCDIR) does not exist

jdart1 committed:
Correct statement about thread safety for probe_root* functions.
Fixes issue from PR#10.

spinojara committed:
Fix race condition caused by init_table failure

jdart1 committed:
make probe_table static

jdart1 committed:
Merge pull request #11 from spinojara/master

Fix race condition caused by init_table failure

MrBrain295 committed:
Fix typo in `tbprobe.h`

folkertvanheusden committed:
tb_init verifies if already initialized by checking if pieceEntry is
NULL. tb_free DID free that memory but did not set the variable to NULL
hence confusing the init part.

jdart1 committed:
Merge pull request #12 from MrBrain295/patch-2

Fix typo in `tbprobe.h`

jdart1 committed:
Merge pull request #13 from folkertvanheusden/tb_free

Re-iniitalize pawnEntry and pieceEntry variables to NULL after free, to ensure initialization check in tb_init will detect state as uninitialized if called again.

cristivlas committed:
customize assert, do not define bool if C++, fix typedef typo

cristivlas committed:
fix error message

cristivlas committed:
fix c++20 crashes

cristivlas committed:
revert non-essential changes

jdart1 committed:
Merge pull request #16 from cristivlas/fix-c++20-crash

Fix c++20 issue when allocation/deallocating pawnEntry and pieceEntry

jdart1 committed:
port stdendian fix from https://github.com/jdart1/arasan-chess

jdart1 committed:
Update copyright. Fix a README typo.

jdart1 committed:
Fix possible integer alignment issues
